pkg/registry/entry.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"net"
+	"net/url"
+	"strings"
+
+	glob "github.com/ryanuber/go-glob"
+)
+
+type Entry interface {
+	Contains(e Entry) bool
+	Match(e string) bool
+	Equal(other Entry) bool
+	String() string
+
+	IsCIDR() bool
+	IsURL() bool
+}
+
+type URLEntry interface {
+	Entry
+	URL() *url.URL
+}
+
+func ParseEntry(s string) Entry {
+	_, ipnet, err := net.ParseCIDR(s)
+	if err == nil {
+		return &cidrEntry{ipnet: ipnet}
+	}
+
+	if u := parseURL(s); u != nil {
+		return &urlEntry{u: u}
+	}
+
+	return nil
```

Comment:
Maybe we should have an InvalidEntry to identify entries that are neither URL nor CIDR.